[BUG] ProtoConversionUtil$AvroSupport static init fails under Avro 1.12 with 'Invalid default for field proto_bytes: ""'

### Bug Description

**What happened:**
`org.apache.hudi.utilities.sources.helpers.ProtoConversionUtil$AvroSupport` fails at class initialization under Avro 1.12.0 with:

```
org.apache.avro.AvroTypeException: Invalid default for field proto_bytes: "" not a "bytes"
```

Because the failure is in a static initializer, the JVM caches it and every subsequent class access in the same JVM throws `NoClassDefFoundError: Could not initialize class org.apache.hudi.utilities.sources.helpers.ProtoConversionUtil$AvroSupport`. Any downstream service that loads this class (e.g. a DeltaStreamer pipeline using `ProtoClassBasedSchemaProvider`) is permanently broken until restart, and even restart doesn't help if Avro 1.12 is on the classpath.

**What you expected:**
The class should initialize under Avro 1.12 the same way it does under Avro 1.11.x.

**Steps to reproduce:**
1. Set Avro 1.12.0 on the classpath (either by overriding `avro.version`, or via transitive dependency from a caller).
2. Load `ProtoConversionUtil$AvroSupport` (any reference to it triggers static init).
3. `ExceptionInInitializerError` at `RECURSION_OVERFLOW_SCHEMA` construction, caused by `AvroTypeException`.

### Environment

**Hudi version:** master (Avro still pinned 1.11.4 in Hudi's pom.xml, so default builds don't hit it; downstream consumers bumping Avro to 1.12+ do)
**Query engine:** N/A — fails at class load, before any query runs
**Relevant configs:** none — triggered purely by classpath

### Logs and Stack Trace

```
java.lang.ExceptionInInitializerError
  at org.apache.hudi.utilities.sources.helpers.ProtoConversionUtil$AvroSupport.<clinit>
  ...
Caused by: org.apache.avro.AvroTypeException: Invalid default for field proto_bytes: "" not a "bytes"
  at org.apache.avro.Schema.validateDefault(Schema.java:1681)
  at org.apache.avro.Schema$Field.<init>(Schema.java:539)
  ...
```

### Root Cause

In `hudi-utilities/src/main/java/org/apache/hudi/utilities/sources/helpers/ProtoConversionUtil.java`:

```java
private static final Schema RECURSION_OVERFLOW_SCHEMA = Schema.createRecord(
    "recursion_overflow", null, "org.apache.hudi.proto", false,
    Arrays.asList(
        new Schema.Field(OVERFLOW_DESCRIPTOR_FIELD_NAME, STRING_SCHEMA, null, ""),
        new Schema.Field(OVERFLOW_BYTES_FIELD_NAME, Schema.create(Schema.Type.BYTES), null, getUTF8Bytes("")) // <-- line 160
    ));
```

The default argument `getUTF8Bytes("")` produces `new byte[0]`. Avro 1.12's `Schema.validateDefault` no longer accepts `byte[]` as a valid BYTES default — it now requires a `String` (which Avro interprets as ISO-8859-1 bytes, matching its canonical JSON form for BYTES defaults). Under 1.11.x the validator was lenient and accepted `byte[]`.

### Fix

Two lines: use `""` (empty string) instead of `getUTF8Bytes("")`, and drop the now-unused static import. The default value is never read at runtime (always overwritten at `ProtoConversionUtil.java:365` by `ByteBuffer.wrap(messageValue.toByteArray())`), and `""` serializes bit-for-bit identically to `new byte[0]` on the wire under Avro 1.11 (both produce `"default":""` in the JSON schema), so this is strictly equivalent behavior while satisfying 1.12's stricter validator.

Rejected alternatives (from experimental testing against both Avro versions):

| Default argument       | Avro 1.11.x | Avro 1.12.0 |
|------------------------|-------------|-------------|
| `new byte[0]` (current)| OK          | FAIL        |
| `""` (fix)             | OK          | OK          |
| `ByteBuffer.wrap(...)` | OK          | FAIL (`Unknown datum class: HeapByteBuffer`) |
| `null`                 | OK          | OK, but strips the default entirely (changes on-disk schema shape) |

PR incoming.

--- a/hudi-utilities/src/main/java/org/apache/hudi/utilities/sources/helpers/ProtoConversionUtil.java
+++ b/hudi-utilities/src/main/java/org/apache/hudi/utilities/sources/helpers/ProtoConversionUtil.java
@@ -65,7 +65,6 @@ import java.util.concurrent.ConcurrentHashMap;
 
 import static org.apache.hudi.common.util.ConfigUtils.getBooleanWithAltKeys;
 import static org.apache.hudi.common.util.ConfigUtils.getIntWithAltKeys;
-import static org.apache.hudi.common.util.StringUtils.getUTF8Bytes;
 import static org.apache.hudi.utilities.config.ProtoClassBasedSchemaProviderConfig.PROTO_SCHEMA_MAX_RECURSION_DEPTH;
 import static org.apache.hudi.utilities.config.ProtoClassBasedSchemaProviderConfig.PROTO_SCHEMA_TIMESTAMPS_AS_RECORDS;
 import static org.apache.hudi.utilities.config.ProtoClassBasedSchemaProviderConfig.PROTO_SCHEMA_WRAPPED_PRIMITIVES_AS_RECORDS;
@@ -142,7 +141,7 @@ public class ProtoConversionUtil {
     private static final String OVERFLOW_BYTES_FIELD_NAME = "proto_bytes";
     private static final HoodieSchema RECURSION_OVERFLOW_SCHEMA = HoodieSchema.createRecord("recursion_overflow", null, "org.apache.hudi.proto", false,
         Arrays.asList(HoodieSchemaField.of(OVERFLOW_DESCRIPTOR_FIELD_NAME, STRING_SCHEMA, null, ""),
-            HoodieSchemaField.of(OVERFLOW_BYTES_FIELD_NAME, HoodieSchema.create(HoodieSchemaType.BYTES), null, getUTF8Bytes(""))));
+            HoodieSchemaField.of(OVERFLOW_BYTES_FIELD_NAME, HoodieSchema.create(HoodieSchemaType.BYTES), null, "")));
     // A cache of the proto class name paired with whether wrapped primitives should be flattened as the key and the generated avro schema as the value
     private static final Map<SchemaCacheKey, HoodieSchema> SCHEMA_CACHE = new ConcurrentHashMap<>();
     // A cache with a key as the pair target avro schema and the proto descriptor for the source and the value as an array of proto field descriptors where the order matches the avro ordering.

